docs/components-and-styling.md

```diff
@@ -55,49 +55,47 @@ It is a good idea to wrap 3rd party components as well in order to adapt them to
 
 ## Component libraries
 
-Every project requires some UI components such as modals, tabs, sidebars, menus, etc. Instead of building those from scratch, you might want to use some of the existing, battle-tested component libraries.
+Every project requires some UI components such as modals, tabs, sidebars, menus, etc.
 
-#### Fully featured component libraries:
+#### Kiwi: our own UI component library:
 
-These component libraries come with their components fully styled.
+Kiwi is our **UI components package**, it is built on **Stitches** api.
+It is a fully **typed** package with Typescript.
+Accessibility standard in these components is bring by **Radix UI** primitives components.
 
-- [Chakra UI](https://chakra-ui.com/) - great library with probably the best developer experience, allows very fast prototyping with decent design defaults. Plenty of components that are very customizable and flexible with accessibility already configured out of the box.
+1. A component must expose an API in accordance with the responsibilities its name suggests, making its purpose obvious.
+   i.e: A `<Text />` component might expose props to modify its font weight or its alignment, not for defining his margin.
+2. If a component exposes props related to CSS, it must allow only theme tokens if any.
+   i.e: A `fontWeight` props on a `<Text />` component should expose theme tokens like `$bold` or `$light` . Raw values like `700` must be forbidden.
+3. A low-level component must expose stitches’s `css` prop to be able to override underlaying css, or to pass contextual style.
+4. A component should not cover every edge cases, modifying `letterSpacing` is not common and therefore should not be exposed on a `<Text />` component.
 
-- [AntD](https://ant.design/) - another great component library that has a lot of different components. Best suitable for creating admin dashboards. However, it might be a bit difficult to change the styles in order to adapt them to a custom design.
+```javascript
+<Text
```

Comment:
I'm not convinced by this syntax.
I find that this does not allow good readability of the code.
For example :
![Capture d’écran 2022-05-11 à 09 48 45](https://user-images.githubusercontent.com/15956476/167797291-f77af217-44b5-4c34-87ec-945624674e3e.png)
Here we see a dialogue with two texts.
I find it a shame to decode what these texts are thanks to the translation, and the assigned css.
So we could have either:
```

const Title = styled('h3', {
  .......
 });
 const Description = styled('p', {
  .......
 });
 
 .....

<Title>
  {t('new.review.sidebar.versions.new.dialog.title')}
</Title>
<Description>
  {t('new.review.sidebar.versions.new.dialog.subtitle')}
</Description>
...
```
Or maybe create others Text Elements. 
For me the inline css makes reading more complex. We go into the details of each component, whereas what we want on first reading is to know what elements the component displays. 
It's the same for `Box`, `Flex` .

Also, the text wraps behind a div. What does not allow to obtain an optimal semantic HTML.

What do you think of my comment?

Reply:
We might discuss and act about this kind of inline vs components pattern if needed.
This pull request is more on writing our current and applied guidelines, having our basis ;D

Reply:
I agree with your points, especially for the semantic.

But maybe `Dialog` was not the ideal exemple 🙂, because we have a `Dialog.Description`, `Dialog.Header` and `Dialog.Title` that do help for the readability.

I also agree that the usage of abstract component like `Text`, `Box` and `Flex` can be confusing and do not help for a great semantic (the three components are `div` by default). But we are beginning to use the prop `as` to override the default element.

```tsx
<Flex as="ul"></Flex>
```

In the end, for me this is not a black & white problem. We should have both in the codebase. 
Maybe we could say that we always begin with a component like `Text`, `Box` and `Flex` with the `as` prop if necessary. And if we feel that the readability of the component is impacted by too many `Text`, `Box` or `Flex`, then maybe we can use `styled()`.